src/Data/ArrayBuffer/ArrayBuffer.js

```diff
@@ -6,31 +6,41 @@ exports.create = function(s) {
   return function () {
     return new ArrayBuffer(s);
   };
-}
+};
 
 exports.byteLength = function(a) {
   return a.byteLength;
-}
+};
 
 exports.sliceImpl = function(s, e, a) {
   return function () {
     return a.slice(s, e);
   };
-}
+};
 
 exports.fromArray = function(s) {
   return (new Uint8Array(s)).buffer;
-}
+};
 
 exports.fromIntArray = function(s) {
   return (new Uint8Array(s)).buffer;
-}
+};
 
 exports.fromString = function(s) {
-  var l = s.length;
-  var ab = new ArrayBuffer(l * 2);
-  var a = new Uint16Array(ab);
-  for (var i = 0; i < l; i++)
-    a[i] = s.charCodeAt(i);
-  return a.buffer;
-}
+  var buf = new ArrayBuffer(s.length*2);
+  var bufView = new Uint16Array(buf);
+  for (var i=0, strLen=s.length; i<strLen; i++) {
+    bufView[i] = s.charCodeAt(i);
+  }
+ return buf;
+};
+
+exports.decodeToString = function(buffer) {
+  const uintBuffer = new Uint16Array(buffer);
+  const reducer = function(accum, point) {
+      // use concat instead of es6 syntax for compatibility
+      return accum.concat([String.fromCharCode(point)]);
+  };
+  const points = uintBuffer.reduce(reducer, new Array());
+  return points.join("");
+};
```

Comment:
I found this one-liner at https://stackoverflow.com/questions/6965107/converting-between-strings-and-arraybuffers 
```
function ab2str(buf) {
  return String.fromCharCode.apply(null, new Uint16Array(buf));
}```

Reply:
I can take a look again in a bit, but I did previously have that and the isomorphism test was failing.  It was a week or more ago so I can't remember the reason for abdandoning that one-liner.

Reply:
I reverted back to the one-liner, as it seems our isomorphism does indeed pass.